packages/cli/tests/e2e/samples/ProvisionAdaptiveCard.tests.ts

```diff
@@ -9,51 +9,29 @@ import { expect } from "chai";
 import fs from "fs-extra";
 import path from "path";
 import { it } from "@microsoft/extra-shot-mocha";
-import {
-  getTestFolder,
-  cleanUp,
-  setSimpleAuthSkuNameToB1Bicep,
-  getSubscriptionId,
-  readContextMultiEnv,
-  readContextMultiEnvV3,
-  getUniqueAppName,
-} from "../commonUtils";
+import { getTestFolder, cleanUp, readContextMultiEnvV3, getUniqueAppName } from "../commonUtils";
 import { BotValidator } from "../../commonlib";
 import { TemplateProject } from "../../commonlib/constants";
-import { CliHelper } from "../../commonlib/cliHelper";
+import { Executor } from "../../utils/executor";
 import { environmentManager } from "@microsoft/teamsfx-core/build/core/environment";
 import { isV3Enabled } from "@microsoft/teamsfx-core";
 describe("teamsfx new template", function () {
   const testFolder = getTestFolder();
-  const subscription = getSubscriptionId();
   const appName = getUniqueAppName();
   const projectPath = path.resolve(testFolder, appName);
   const env = environmentManager.getDefaultEnvName();
   let teamsAppId: string | undefined;
 
   it(`${TemplateProject.AdaptiveCard}`, { testPlanCaseId: 15277474 }, async function () {
-    if (isV3Enabled()) {
-      await CliHelper.openTemplateProject(appName, testFolder, TemplateProject.AdaptiveCard);
-      expect(fs.pathExistsSync(projectPath)).to.be.true;
-      expect(fs.pathExistsSync(path.resolve(projectPath, "infra"))).to.be.true;
-    } else {
-      await CliHelper.createTemplateProject(appName, testFolder, TemplateProject.AdaptiveCard);
-      expect(fs.pathExistsSync(projectPath)).to.be.true;
-      expect(fs.pathExistsSync(path.resolve(projectPath, ".fx"))).to.be.true;
-    }
+    await Executor.openTemplateProject(appName, testFolder, TemplateProject.AdaptiveCard);
+    expect(fs.pathExistsSync(projectPath)).to.be.true;
+    expect(fs.pathExistsSync(path.resolve(projectPath, "infra"))).to.be.true;
 
     // Provision
-    if (isV3Enabled()) {
-    } else {
-      await setSimpleAuthSkuNameToB1Bicep(projectPath, env);
-      await CliHelper.setSubscription(subscription, projectPath);
-    }
-    await CliHelper.provisionProject(projectPath);
+    await Executor.provision(projectPath);
```

Comment:
Do you want to check the result of the commands? If the command failed to run, the `result.success` will be `false`.

Reply:
Can I add the such as following code to check the failure cause？
```
    const { success, stderr } = await Executor.provision(projectPath);
    if (!success) {
      console.log(stderr);
      chai.assert.fail("Provision failed");
    }
```

Reply:
E2E: https://github.com/OfficeDev/TeamsFx/actions/runs/4954818966/jobs/8864063420